### DISCUSSION TITLE
Some missing events that I scheduled

### DISCUSSION THREAD

**Discussion by musicaljg** [derekantrican/GAS-ICS-Sync#420] [Q&A] [answered]
The GAS-ICS-Sync is generally working very well for me at displaying my Outlook (work) events on my Google (personal) calendar.

Occasionally, Outlook events just don't show up on the Google Calendar, even though there are no errors in the sync logs.

I've noticed that the events that don't sync are most often ones that I've created. (For example, a meeting I've scheduled with other colleagues attending.) 

It's not every event that I've created. There's not anything that clearly distinguishes the events I created that sync properly from the ones that don't (e.g., it's not just recurring or one-time events).

Has anyone else experienced this? Any ideas? Thank you.

**Comment by jonas0b1011001** [comment]
We'd need the ical definition of one of the events that do not sync, to see if there is anything odd.
So once you notice this behavior again, manually download the ics file, open it in any text editor and search for the event title that is not getting synced. Share the `BEGIN:VEVENT...END:VEVENT` block of that event, remove sensitive data as needed.

**Comment by musicaljg** [comment]
Thank you. Here it is:

BEGIN:VEVENT
ORGANIZER;CN=[organizer removed]
ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE;CN=[attendee 1 removed]
ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE;CN=[attendee 2 removed]
DESCRIPTION:[message removed]\n\n_______________________________________________
 _________________________________\nMicrosoft Teams meeting\nJoin on your c
 omputer\, mobile app or room device\nClick here to join the meeting<[Teams meeting details removed]_________________________________________________________________
 _______________\n
UID:040000008200E00074C5B7101A82E0080000000018A0915EF46FDA01000000000000000
 0100000008C3B8A5F9C21AC4BB7C80B5BF6833B09
SUMMARY:[meeting name removed]
DTSTART:20240307T220000Z
DTEND:20240307T223000Z
CLASS:PUBLIC
PRIORITY:5
DTSTAMP:20240306T185354Z
TRANSP:OPAQUE
STATUS:CONFIRMED
LOCATION:Microsoft Teams Meeting
X-MICROSOFT-CDO-BUSYSTATUS:BUSY
X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
X-MICROSOFT-CDO-ALLDAYEVENT:FALSE
X-MICROSOFT-CDO-IMPORTANCE:1
X-MICROSOFT-CDO-INSTTYPE:0
X-MICROSOFT-SKYPETEAMSMEETINGURL:[Teams meeting URL removed]
X-MICROSOFT-ONLINEMEETINGEXTERNALLINK:
X-MICROSOFT-ONLINEMEETINGINFORMATION:{"OnlineMeetingChannelId":null\,"Onlin
 eMeetingProvider":3}
X-MICROSOFT-DONOTFORWARDMEETING:FALSE
X-MICROSOFT-DISALLOW-COUNTER:FALSE
X-MICROSOFT-LOCATIONDISPLAYNAME:Microsoft Teams Meeting
BEGIN:VALARM
DESCRIPTION:REMINDER
TRIGGER;RELATED=START:-PT10M
ACTION:DISPLAY
END:VALARM
END:VEVENT

**Comment by jonas0b1011001** [comment]
Hmm that looks normal and should sync just fine.

What version of the script are you running?
Have you done any modifications apart from the settings in code.gs?

**Comment by musicaljg** [comment]
This is what I'm running (with my work calendar redacted): https://script.google.com/d/1t0rn4OSLBAmgJ-KUk7jo6q93NRfTFvzUx3BXic5upTFBBc4xdFhuke1M/edit?usp=sharing

I think I copied it sometime last year because there was a bug for which you provided a fix in comments but that wasn't yet merged into the main project.  I think any modifications would have been tied to that, but don't recall the details.

**Reply by musicaljg** [reply]
@jonas0b1011001 Any luck determining what might be happening here? Thank you.

**Reply by jonas0b1011001** [reply]
Sorry, i don't have access to the file you posted. Can you change permission to 'anyone with the link'?

**Reply by musicaljg** [reply]
Updated - thanks

**Reply by jonas0b1011001** [reply]
The script still contains the url to an outlook calendar, not sure if you want to remove that.

---
Unfortunately I don't see any obvious reason for the missing events.
Are there events missing currently? If so, have you tried syncing the ics to a different target calendar to see if they show up there?

**Reply by musicaljg** [reply]
Thank you. No worries. I'll probably try using the latest version of the script when I have a chance.

**Reply by musicaljg** [reply, marked as answer]
Using the latest version seems to have resolved this. I must have messed something up in attempting to customize it in some way. Thank you for your help.
